src/tracer/MARBL_tracers.F90

```diff
@@ -0,0 +1,734 @@
+!! call MARBL's initialization routine
+subroutine configure_MARBL_tracers(GV, param_file, CS)
+  type(verticalGrid_type),    intent(in) :: GV   !< The ocean's vertical grid structure
+  type(param_file_type),      intent(in) :: param_file !< A structure to parse for run-time parameters
+  type(MARBL_tracers_CS),     pointer    :: CS   !< A pointer that is set to point to the control
+                                                 !! structure for this module
+
+#include "version_variable.h"
+  character(len=40)  :: mdl = "MARBL_tracers" ! This module's name.
+  character(len=256) :: log_message
+  character(len=256) :: marbl_in_line(1)
+  integer :: m, nz, marbl_settings_in, read_error
+  nz = GV%ke
+  marbl_settings_in = 615
+
+  ! (1) Read all relevant parameters and write them to the model log.
+  call log_version(param_file, mdl, version, "")
+  call get_param(param_file, mdl, "marbl_settings_file", CS%marbl_settings_file, &
+                 "The name of a file from which to read the run-time "//&
+                 "settings for MARBL.", default="marbl_in")
+  call get_param(param_file, mdl, "atm_co2_const", CS%atm_co2_const, &
+                 "Value to send to MARBL as xco2", default=284.317)
+  call get_param(param_file, mdl, "atm_alt_co2_const", CS%atm_alt_co2_const, &
+                 "Value to send to MARBL as xco2_alt_co2", default=284.317)
+
+  ! (2) Read marbl settings file and call put_setting()
+
+  ! (2a) only master task opens file
+  if (is_root_PE()) &
+     ! read the marbl_in into buffer
+     open(unit=marbl_settings_in, file=CS%marbl_settings_file, iostat=read_error)
+  call broadcast(read_error, root_PE())
+  if (read_error .ne. 0) then
+    write(log_message, '(A, I0, 2A)') "IO ERROR ", read_error, &
+         "opening namelist file : ", trim(CS%marbl_settings_file)
+    call MOM_error(FATAL, log_message)
+  end if
+
+  ! (2b) master task reads file and broadcasts line-by-line
+  marbl_in_line = ''
+  do
+    ! i. Read next line on master, iostat value out
+    !    (Exit loop if read is not successful; either read error or end of file)
+    if (is_root_PE()) read(marbl_settings_in, "(A)", iostat=read_error) marbl_in_line(1)
+    call broadcast(read_error, root_PE())
+    if (read_error .ne. 0) exit
+
+    ! ii. Broadcast line just read in on root PE to all tasks
+    call broadcast(marbl_in_line, 256, root_PE())
+
+    ! iii. All tasks call put_setting (TODO: openMP blocks?)
+    call marbl_instances%put_setting(marbl_in_line(1))
+  end do
+
+  ! (2c) we should always reach the EOF to capture the entire file...
+  if (.not. is_iostat_end(read_error)) then
+     write(log_message, '(3A, I0)') "IO ERROR reading ", trim(CS%marbl_settings_file), ": ", read_error
+     call MOM_error(FATAL, log_message)
+  else
+     if (is_root_PE()) then
+       ! TODO: Better way to get message into log?
+       write(*, '(3A)') "Read '", trim(CS%marbl_settings_file), "' until EOF."
```

Comment:
You can use `call MOM_error(NOTE, "Whatever message")`. This won't actually cause the model to crash but is just used for informational messages.